src/azul/drs.py

```diff
@@ -209,38 +237,61 @@ class Client:
+
+        For 'https', the request response containing the object will be
+        returned.
+
+        for 'gs', the gs:// url for the object will be returned.
+        """
         if access_id is None:
-            return self._get_object(object_id)
+            return self._get_object(object_id, access_method=access_method)
         else:
-            return self._get_object_access(object_id, access_id)
+            return self._get_object_access(object_id, access_id, access_method=access_method)
 
-    def _get_object(self, object_id: str) -> requests.Response:
+    def _get_object(self, object_id: str, access_method: AccessMethod) -> Union[requests.Response, str]:
         url = self.url / object_id
         while True:
             response = requests.get(url)
             if response.status_code == 200:
                 # Bundles are not supported therefore we can expect 'access_methods'
                 access_methods = response.json()['access_methods']
-                assert len(access_methods) == 1, 'Only one access method is supported currently'
-                for access_method in access_methods:
-                    if 'access_id' in access_method:
-                        return self._get_object_access(object_id, access_method['access_id'])
-                    elif 'access_url' in access_method:
-                        # We only support https as an access type
-                        assert access_method['type'] == 'https'
-                        access_url_ = access_method['access_url']['url']
+                method = one(m for m in access_methods if m['type'] == access_method.value)
+                if 'access_id' in method:
+                    return self._get_object_access(object_id, method['access_id'],
+                                                   access_method=access_method)
+                elif 'access_url' in method:
+                    access_url_ = method['access_url']['url']
+                    if method['type'] == 'https':
                         return requests.get(access_url_)
+                    elif method['type'] == 'gs':
+                        return access_url_
                     else:
                         assert False
+                else:
+                    assert False
             elif response.status_code == 202:
                 # note the busy wait
                 wait_time = int(response.headers['retry-after'])
                 time.sleep(wait_time)
             else:
                 raise DRSError(response.status_code, response.text)
 
-    def _get_object_access(self, object_id: str, access_id: str) -> requests.Response:
+    def _get_object_access(self,
+                           object_id: str,
+                           access_id: str,
+                           access_method: AccessMethod
+                           ) -> Union[requests.Response, str]:
+        """
+        If url_type is 'gs', then the gs:// URL is returned, otherwise, for
```

Comment:
url_type doesn't exist and access_method values aren't strings. Please be sure to check all of your documentation for thsi type of incosnistency